google/cloud/storage/oauth2/authorized_user_credentials.h

```diff
@@ -92,16 +115,23 @@ class AuthorizedUserCredentials : public Credentials {
 
     // TODO(#516) - use retry policies to refresh the credentials.
     auto response = request_.MakeRequest(payload_);
-    if (200 != response.status_code) {
+    if (response.status_code >= 300) {
+      return false;
+    }
+    nl::json access_token = nl::json::parse(response.payload, nullptr, false);
+    if (access_token.is_discarded() or access_token.count("token_type") == 0U or
+        access_token.count("access_token") == 0U or
+        access_token.count("id_token") == 0U or
+        access_token.count("expires_in") == 0U) {
       return false;
     }
-    nl::json access_token = nl::json::parse(response.payload);
     std::string header = "Authorization: ";
-    header += access_token["token_type"].get_ref<std::string const&>();
+    header += access_token.value("token_type", "");
```

Comment:
We should be validating above that the these fields are both present and not empty -- having an empty value here makes no sense.

Reply:
I refactored the parsing to a function, let me know what you think.